barretenberg/cpp/src/barretenberg/stdlib/primitives/uint/logic.cpp

```diff
@@ -281,11 +281,25 @@ uint<Builder, Native> uint<Builder, Native>::logic_operator(const uint& other, c
 
     ReadData<field_t<Builder>> lookup;
     if (op_type == XOR) {
-        lookup = plookup_read<Builder>::get_lookup_accumulators(
-            MultiTableId::UINT32_XOR, field_t<Builder>(*this), field_t<Builder>(other), true);
+        if constexpr (std::is_same_v<Native, uint64_t>) {
+            // use the 64-bit XOR lookup
+            lookup = plookup_read<Builder>::get_lookup_accumulators(
```

Comment:
I'd add a static assert here or in class definition to ensure that 32-bit and 64-bit uints are the only available options

Reply:
Well for `uint8_ct, uin16_ct, uint32_ct` the 32-bit lookup works fine. So previously everything for these three types worked. But `uint64_ct` operations obviously failed with 32-bit lookup tables. Adding this explanation with examples for clarity.